analyses/doublet-detection/scripts/01b_detect-doublets.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env python3
+
+# This script runs scrublet on a given AnnData file and exports results as a TSV.
+
+
+import argparse
+import anndata
+import scrublet
+import pandas
+from pathlib import Path
+import sys
+
+# Function to run scrublet
+def run_scrublet(adata: anndata.AnnData, random_seed: int) -> pandas.DataFrame:
+    """
+        Run scrublet on a counts matrix from an AnnData object, following https://github.com/swolock/scrublet?tab=readme-ov-file#quick-start
+        Returns a dataframe with the following columns:
+            - `barcodes` is the droplet's unique barcode
+            - `scrublet_score` is the probability that each droplet is a doublet
+            - `scrublet_prediction` is the predicted droplet status ("doublet" or "singlet"), based on an automatically-set threshold
+    """
+
+    scrub = scrublet.Scrublet(adata.X, random_state = random_seed)
+    # predicted_doublets is boolean array, where True are predicted doublets and False are predicted singlets
+    doublet_scores, predicted_doublets = scrub.scrub_doublets()
+
+    # convert True/False to string values
+    predicted_doublets_str = [("doublet" if x else "singlet") for x in predicted_doublets]
+
+    results = {"barcodes" : adata.obs_names, "scrublet_score" : doublet_scores, "scrublet_prediction" : predicted_doublets_str}
+    results_df = pandas.DataFrame(results)
+
+    return(results_df)
+
+def main() -> None:
+
+    parser = argparse.ArgumentParser(
+        description="Detect doublets on a set of AnnData objects using scrublet.",
+    )
+    parser.add_argument(
+        "--dataset_name",
+        type=str,
+        default="",
```

Comment:
You may want to set this as `required=True` rather than setting the default like this?